### ISSUE TITLE
estimates.detrend_df_f use_residuals keyword has no effect

### ISSUE THREAD

**Issue by EricThomson** [flatironinstitute/CaImAn#1188]
In [estimates.detrend_df_f()](https://github.com/flatironinstitute/CaImAn/blob/9b0b79ca61f20ce93259b9833e1fe18e26d4e086/caiman/source_extraction/cnmf/estimates.py#L777)  , there is a keyword argument `use_residuals` that calls `detrend_df_f()`

```
        if use_residuals:
            if self.R is None:
                if self.YrA is None:
                    R = None
                else:
                    R = self.YrA
            else:
                R = self.R
        else:
            R = None

        self.F_dff = detrend_df_f(self.A, self.b, self.C, self.f, self.YrA,
                                  quantileMin=quantileMin,
                                  frames_window=frames_window,
                                  flag_auto=flag_auto, use_fast=use_fast,
                                  detrend_only=detrend_only)
```
Clearly the intent was to set `R` (residuals) and use that as a replacement for `self.YrA`, but instead `self.YrA` (the value of residuals) was never changed, so `use_residuals` has no effect. This should be an easy fix. Just raising the issue here first.

**Comment by EricThomson** [comment]
Note I fixed in CNMF notebook update that I should push soon (when I get back from conference) and it is working as expected.

**Comment by vncntprvst** [comment]
Just catching up with this issue. This is intriguing. Looking at the git blame, the `estimates.py`'s call to `detrend_df_f` used to have `R` as argument, when @epnev moved the Estimates class and related methods in that [commit ](https://github.com/flatironinstitute/CaImAn/commit/60d30f95a8e104079a7f34ec8f53d80409c4ee4c) five years ago:
```
        self.F_dff = detrend_df_f(self.A, self.b, self.C, self.f, R,
                                  quantileMin=quantileMin,
                                  frames_window=frames_window,
                                  flag_auto=flag_auto, use_fast=use_fast)
```
Then he changed `R` to `self.YrA` [a few weeks later](https://github.com/flatironinstitute/CaImAn/commit/0e1bb6cb4602ed5450d86a9e12de6d1579c0e8f8) that year. 
This is weird because it looks like this a potential reason for the df/f not being denoised (referenced in https://github.com/nel-lab/mesmerize-core/issues/260 above), but we used to be able to get a denoised df/f more recently than 2018.

**Comment by EricThomson** [comment]
Yikes, unfortunate we didn't test for that. It should be fixed in the next release (the fix is buried in the PR for the CNMF notebook which should be out in the next release).

**Comment by vncntprvst** [comment]
Looking forward to that new release. Thanks for all the work on this!

**Comment by EricThomson** [comment]
Merged into dev (#1075 ) will be in next release.

**Comment by vncntprvst** [comment]
Yay! 
Thank you
